### PR TITLE
Include source generators in TryGetMethodDescriptorAsync

### LINKED ISSUE #51633: CodeLens references unhandled exception
**Version Used**: Version 16.10.0 Preview 1.0 (but I believe it is the same in 16.9 which was released yesterday).

**Steps to Reproduce**:

1. Clone https://github.com/pakrym/jab/commit/3b5eaefeb0142f560d52cb653f0400be750888f8
2. Open the project solution
3. Open `jab\src\samples\ConsoleSample\Program.cs` file
4. You should see an exception:

image

**Expected Behavior**:

No exception.

**Actual Behavior**:

Exception.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #51633

### CODE REVIEW

File: src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs [hunk 1]
```diff
@@ -246,7 +246,7 @@ private static async Task<ReferenceMethodDescriptor> TryGetMethodDescriptorAsync
                 return null;
             }
 
-            var document = solution.GetDocument(doc.Id);
+            var document = await solution.GetDocumentAsync(doc.Id, includeSourceGenerated: true, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs:249]: i honestly don't understand the point of the before/after line in teh first place.  Right above this, we already found teh 'doc', so why not just use that?

Comment (Youssef1313) [reply src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs:249]: @CyrusNajmabadi Honestly, me too. Let me delete it and see if there are any failures that explains this.

**Review by CyrusNajmabadi — APPROVED** [review]
@jasonmalinowski any SG concerns here?

File: src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs [hunk 2]
```diff
@@ -240,13 +240,12 @@ private static ISymbol GetEnclosingMethod(SemanticModel semanticModel, Location
...
         {
-            var doc = solution.GetDocument(commonLocation.SourceTree);
-            if (doc == null)
+            var document = solution.GetDocument(commonLocation.SourceTree);
+            if (document == null)
             {
                 return null;
             }
 
-            var document = solution.GetDocument(doc.Id);
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs:249]: I am terrified to wonder how this code ever happened in the first place.

**Review by jasonmalinowski — APPROVED** [review]
:shipit:

File: src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs]: I'm guessing this got refactored over the years and ended up in this form without anybody realizing it.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi My only concern is the that the original code ever existed in our codebase in the first place.

File: src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs]: This goes back to https://github.com/dotnet/roslyn/commit/b83e57778fda657b1b1b6f7ddc5c92cecaac3036

**Comment by jasonmalinowski** [comment]
Merging this so it's in this at least in going forward; we're still having a conversation about a backport for 16.9.

**Pull request merged**